eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/PoolReap.hs

```diff
@@ -113,24 +115,35 @@ instance
 
 poolReapTransition ::
   forall era.
-  HasField "_poolDeposit" (PParams era) Coin =>
   TransitionRule (ShelleyPOOLREAP era)
 poolReapTransition = do
-  TRC (pp, PoolreapState us a ds ps, e) <- judgmentContext
+  TRC (_pp, PoolreapState us a ds ps, e) <- judgmentContext
 
-  let retired :: Set (KeyHash 'StakePool (EraCrypto era))
+  let -- The set of pools retiring this epoch
+      retired :: Set (KeyHash 'StakePool (EraCrypto era))
       retired = eval (dom (psRetiring ps ▷ setSingleton e))
-      pr :: Map.Map (KeyHash 'StakePool (EraCrypto era)) Coin
-      pr = Map.fromSet (const (getField @"_poolDeposit" pp)) retired
+      -- The Map of pools (retiring this epoch) to their deposits
+      retiringDeposits, remainingDeposits :: Map.Map (KeyHash 'StakePool (EraCrypto era)) Coin
+      (retiringDeposits, remainingDeposits) = Map.partitionWithKey (\k _ -> Set.member k retired) (psDeposits ps)
       rewardAcnts :: Map.Map (KeyHash 'StakePool (EraCrypto era)) (RewardAcnt (EraCrypto era))
       rewardAcnts = Map.map ppRewardAcnt $ eval (retired ◁ psStakePoolParams ps)
       rewardAcnts_ :: Map.Map (KeyHash 'StakePool (EraCrypto era)) (RewardAcnt (EraCrypto era), Coin)
-      rewardAcnts_ = Map.intersectionWith (,) rewardAcnts pr
+      rewardAcnts_ = Map.intersectionWith (,) rewardAcnts retiringDeposits
       rewardAcnts' :: Map.Map (RewardAcnt (EraCrypto era)) Coin
       rewardAcnts' =
         Map.fromListWith (<+>)
           . Map.elems
           $ rewardAcnts_
+      {- An alternate approach
+      (refunds,mRefunds) = Map.foldlWithKey accum (mempty,mempty) (eval (retired ◁ psStakePoolParams ps))
+        where accum ans@(refund,unclaim) poolHash poolparam  =
+                case Map.lookup poolHash retiringDeposits of
+                  Nothing -> ans
+                  Just coin -> if eval (cred  ∈ dom (rewards ds))
+                                  then (Map.insertWith (<+>) cred coin refund,unclaim)
+                                  else (refund, Map.insertWith (<+>) cred coin unclaim)
+                    where cred = getRwdCred (_poolRAcnt poolparam)
+      -}
```

Comment:
is there a reason to keep this?

Reply:
Not really,  I thought we might discuss making this change. To me it is really much cleaner